docs/json-map.md

```diff
@@ -0,0 +1,72 @@
+# Overview
+JSON Map is a JSON-based format which describes part of the structure of another JSON document. Maps inform clients of the sizes of parts of the JSON and to allow constant time construction of subsets of the document.
+
+For example, suppose you have a 20GB JSON file which contains a huge array with one million elements along with some surrounding document structure. You can construct a map with a 0.1% target size (20 MB). The map will identify the path to the huge array, the number of elements it has, where each nth array element begins, and where the array itself starts and ends. 
+
+To write a copy of the file with only the first 1,000 array elements, copy the file content up to the array start, copy the range from element 0 to element 999, omit the last comma, and copy the range from the array end to the end of the file.
+
+
+# JSON Map Structure
+A JSON Map consists of a hierarchy of **nodes** which mirror the hierarchical structure of the underlying JSON document. A map is constructed with a specific **target size ratio** (ex: 1%) which determines a size threshold for including items in the map.
```

Comment:
>nodes [](start = 40, length = 5)

It would really help to have an example showing the skeleton of a SARIF file together with the corresponding map file.

Reply:
Added.